tachyon/math/polynomials/univariate/univariate_polynomial.h

```diff
@@ -89,6 +90,24 @@ class UnivariatePolynomial final
     return coefficients_.Evaluate(point);
   }
 
+  constexpr UnivariatePolynomial DivByVanishing(
+      const std::vector<Field>& roots) {
+    UnivariatePolynomial divided_poly = *this;
+    for (const Field& root : roots) {
+      UnivariatePolynomial divisor =
+          UnivariatePolynomial(Coefficients({-root, Field::One()}));
+      divided_poly = divided_poly / divisor;
+    }
+    return divided_poly;
+  }
+
+  constexpr static Field EvaluateVanishingPolyByRoots(
+      const std::vector<Field>& roots, Field point) {
+    return std::accumulate(
+        roots.begin(), roots.end(), Field::One(),
+        [point](Field acc, Field root) { return (point - root) * acc; });
```

Comment:
Considering possible implementation from https://en.cppreference.com/w/cpp/algorithm/accumulate

```c++
template<class InputIt, class T, class BinaryOperation>
constexpr // since C++20
T accumulate(InputIt first, InputIt last, T init, BinaryOperation op)
{
    for (; first != last; ++first)
        init = op(std::move(init), *first); // std::move since C++20
 
    return init;
}
```

You should do like below. Plus, could you apply these patterns to where `std::accumulate` is used?

```suggestion
        [point](Field& acc, const Field& root) { return acc *= (point - root); });
```